### PR TITLE
refactor(ng-dev/pr): move pr configuration out of the merge directory

### PR DESCRIPTION
**Pull request by josephperrott**
As part of #203, refactoring the pr directory into more reusable pieces.

### CODE REVIEW

File: ng-dev/pr/config.ts [hunk 1]
```diff
@@ -6,16 +6,22 @@
...
+export interface GithubApiMergeStrategyConfig {
+  /** Default method used for merging pull requests */
+  default: GithubApiMergeMethod;
+  /** Labels which specify a different merge method than the default. */
+  labels?: {pattern: string; method: GithubApiMergeMethod}[];
+}
+
 /**
  * Configuration for the merge script with all remote options specified. The
  * default `MergeConfig` has does not require any of these options as defaults
```
Comment (devversion) [inline ng-dev/pr/config.ts:27]: This is something I'm not 100% sure about. Does it make sense to capture the merge-specific configuration in something as generic as `ng-dev/pr`? are there any benefits you foresee here?

Comment (josephperrott) [reply ng-dev/pr/config.ts:27]: I actually think we should go a step further and rename this config to PullRequestConfig, while the primary usage is for merging, we rely on it for other actions in the `pr` directory like the `determine-target-branches` (soon to be `info`).  While much of it is about merging PRs, it really it describing how the tooling interacts with PRs.

Additionally, this aligns with how the rest of the ng-dev tooling has a config per primary subcommand (i.e. `ReleaseConfig`, `CaretakerConfig`, `FormatConfig`)

Comment (devversion) [reply ng-dev/pr/config.ts:27]: That's fair. I wish some of the interfaces, like for the API merge strategy to stay local to their logic, but I don't feel strongly

**Review by devversion — APPROVED** [review]
Lgtm we should rename the config

File: ng-dev/pr/config.ts (hunk 1 above)
Comment (josephperrott) [reply ng-dev/pr/config.ts:27]: We can look to move the merge strategy back, I wasn't going to move it initially, but it caused a circular dependency iat the build level.  I will move all the symbols used for the ApiMergeStrategy into the api-merge.ts file

Comment (josephperrott) [reply ng-dev/pr/config.ts:27]: Actually I missed a piece before that comment, we can't avoid the circular dependency

**Comment by josephperrott** [comment]
This PR was merged into the repository by commit cf18f08e546a481b3c0fe5506df587a652946681.

**Pull request closed without merging**